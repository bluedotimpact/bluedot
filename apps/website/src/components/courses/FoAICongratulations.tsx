import React from 'react';
import clsx from 'clsx';
import { CTALinkOrButton } from '@bluedot/ui';

type FoAICongratulationsProps = {
  className?: string;
};

const FoAICongratulations: React.FC<FoAICongratulationsProps> = ({ className }) => {
  return (
    <div className={clsx('congratulations container-lined p-8 bg-white', className)}>
      <div className="flex flex-col gap-2">
        <p className="congratulations__title bluedot-h4 mb-2 text-center">
          Congratulations on completing the Future of AI course!
        </p>
        <p className="congratulations__description bluedot-p">
          Want to connect with others taking AI safety seriously? <span className="font-semibold">Join 3,000+ FoAI course graduates</span> turning knowledge into action.
        </p>
      </div>
      <div className="mt-4 flex justify-start">
        <CTALinkOrButton
          url="https://community.bluedot.org"
          variant="primary"
          target="_blank"
        >
          Join the Community
        </CTALinkOrButton>
      </div>
    </div>
  );
};

export default FoAICongratulations;
