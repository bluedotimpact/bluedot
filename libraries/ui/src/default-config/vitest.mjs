/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
