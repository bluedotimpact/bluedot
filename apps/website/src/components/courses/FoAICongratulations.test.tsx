import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import FoAICongratulations from './FoAICongratulations';

describe('FoAICongratulations', () => {
  test('renders default as expected', () => {
    const { container } = render(
      <FoAICongratulations />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with custom className', () => {
    const { container } = render(
      <FoAICongratulations className="max-w-2xl mx-auto" />,
    );
    expect(container).toMatchSnapshot();
  });
});
