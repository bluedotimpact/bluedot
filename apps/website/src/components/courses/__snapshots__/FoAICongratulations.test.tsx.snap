// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FoAICongratulations > renders default as expected 1`] = `
<div>
  <div
    class="congratulations container-lined p-8 bg-white"
  >
    <div
      class="flex flex-col gap-2"
    >
      <p
        class="congratulations__title bluedot-h4 mb-2 text-center"
      >
        Congratulations on completing the Future of AI course!
      </p>
      <p
        class="congratulations__description bluedot-p"
      >
        Want to connect with others taking AI safety seriously? 
        <span
          class="font-semibold"
        >
          Join 3,000+ FoAI course graduates
        </span>
         turning knowledge into action.
      </p>
    </div>
    <div
      class="mt-4 flex justify-start"
    >
      <a
        class="cta-button flex items-center justify-center transition-all duration-200 w-fit whitespace-nowrap cursor-pointer not-prose disabled:opacity-50 disabled:pointer-events-none aria-disabled:opacity-50 aria-disabled:pointer-events-none text-sm px-4 py-3 rounded-sm font-[650] cta-button--primary bg-bluedot-normal link-on-dark"
        href="https://community.bluedot.org"
        tabindex="0"
        target="_blank"
      >
        Join the Community
      </a>
    </div>
  </div>
</div>
`;

exports[`FoAICongratulations > renders with custom className 1`] = `
<div>
  <div
    class="congratulations container-lined p-8 bg-white max-w-2xl mx-auto"
  >
    <div
      class="flex flex-col gap-2"
    >
      <p
        class="congratulations__title bluedot-h4 mb-2 text-center"
      >
        Congratulations on completing the Future of AI course!
      </p>
      <p
        class="congratulations__description bluedot-p"
      >
        Want to connect with others taking AI safety seriously? 
        <span
          class="font-semibold"
        >
          Join 3,000+ FoAI course graduates
        </span>
         turning knowledge into action.
      </p>
    </div>
    <div
      class="mt-4 flex justify-start"
    >
      <a
        class="cta-button flex items-center justify-center transition-all duration-200 w-fit whitespace-nowrap cursor-pointer not-prose disabled:opacity-50 disabled:pointer-events-none aria-disabled:opacity-50 aria-disabled:pointer-events-none text-sm px-4 py-3 rounded-sm font-[650] cta-button--primary bg-bluedot-normal link-on-dark"
        href="https://community.bluedot.org"
        tabindex="0"
        target="_blank"
      >
        Join the Community
      </a>
    </div>
  </div>
</div>
`;
